app/scripts/modules/core/src/managed/ManagedResourceObject.tsx

```diff
@@ -57,20 +57,37 @@ const getResourceRoutingInfo = (
 
 export const ManagedResourceObject = ({ resource, artifact, depth }: IManagedResourceObjectProps) => {
   const { version: currentVersion, buildNumber: currentBuild } = parseName(artifact?.versions.current || '') || {};
+  const { version: deployingVersion, buildNumber: deployingBuild } =
+    parseName(artifact?.versions.deploying || '') || {};
   const { kind } = resource;
   const resourceName = getResourceName(resource);
   const routingInfo = getResourceRoutingInfo(resource) ?? { state: '', params: {} };
   const route = useSref(routingInfo.state, routingInfo.params);
 
+  const currentPill = artifact?.versions.current ? (
+    <Pill text={currentBuild ? `#${currentBuild}` : currentVersion || artifact.versions.current || 'unknown'} />
+  ) : null;
+  const deployingPill = artifact?.versions.deploying ? (
+    <>
+      <Icon appearance="dark" name="caretRight" size="medium" />
+      <Pill
+        animateOpacity={true}
+        bgColor="#61b9ff"
+        text={deployingBuild ? `#${deployingBuild}` : deployingVersion || artifact.versions.deploying || 'unknown'}
```

Comment:
since we won't show this component if `artifact?.versions.deploying` is false, do we need a fallback to `'unknown'` here?

Reply:
Good call!